pkg/cmd/readyz/readyz.go

```diff
@@ -0,0 +1,199 @@
+	fs.StringVar(&r.servingCertFile, "serving-cert-file", "", "Health probe server TLS client certificate file. (required)")
+	fs.StringVar(&r.servingKeyFile, "serving-key-file", "", "Health probe server TLS client key file. (required)")
+	fs.StringVar(&r.clientCertFile, "client-cert-file", "", "Etcd TLS client certificate file. (required)")
+	fs.StringVar(&r.clientKeyFile, "client-key-file", "", "Etcd TLS client key file. (required)")
+	fs.StringVar(&r.clientCACertFile, "client-cacert-file", "", "Etcd TLS client CA certificate file. (required)")
+}
+
+// Validate verifies the inputs.
+func (r *readyzOpts) Validate() error {
+	if len(r.targetEndpoint) == 0 {
+		return errors.New("missing required flag: --target")
+	}
+	if len(r.servingCertFile) == 0 {
+		return errors.New("missing required flag: --serving-cert-file")
+	}
+	if len(r.servingKeyFile) == 0 {
+		return errors.New("missing required flag: --serving-key-file")
+	}
+	if len(r.clientCertFile) == 0 {
+		return errors.New("missing required flag: --client-cert-file")
+	}
+	if len(r.clientKeyFile) == 0 {
+		return errors.New("missing required flag: --client-key-file")
+	}
+	if len(r.clientCACertFile) == 0 {
+		return errors.New("missing required flag: --client-cacert-file")
+	}
+
+	return nil
+}
+
+// Run contains the logic of the readyz command which checks the health of the etcd member
+func (r *readyzOpts) Run() error {
+	shutdownCtx, cancel := context.WithCancel(context.Background())
+	shutdownHandler := server.SetupSignalHandler()
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/readyz", r.getReadyzHandlerFunc(shutdownCtx))
+	// Handle the /healthz endpoint as well since the static pod controller's guard pods check the /healthz endpoint
+	// https://github.com/openshift/library-go/blob/edab248e63516c65a93467eaa8224c86d69f5de9/pkg/operator/staticpod/controller/guard/manifests/guard-pod.yaml#L44
+	mux.HandleFunc("/healthz", r.getReadyzHandlerFunc(shutdownCtx))
+
+	addr := fmt.Sprintf("0.0.0.0:%d", r.listenPort)
+	klog.Infof("Listening on %s", addr)
+
+	server := &http.Server{
+		Addr:        addr,
+		Handler:     mux,
+		BaseContext: func(_ net.Listener) context.Context { return shutdownCtx },
+	}
+	go func() {
+		defer cancel()
+		<-shutdownHandler
+		klog.Infof("Received SIGTERM or SIGINT signal, shutting down server.")
+		server.Shutdown(shutdownCtx)
+	}()
+
+	c := net.ListenConfig{}
+	c.Control = permitAddressReuse
+	ln, err := c.Listen(shutdownCtx, "tcp", addr)
+	if err != nil {
+		return err
+	}
+	err = server.ServeTLS(ln, r.servingCertFile, r.servingKeyFile)
+	if err == http.ErrServerClosed {
+		err = nil
+	}
+	<-shutdownCtx.Done()
+	return err
+}
+
+func (r *readyzOpts) getReadyzHandlerFunc(ctx context.Context) http.HandlerFunc {
+	return func(w http.ResponseWriter, _ *http.Request) {
+		etcdClient, err := r.newETCD3Client(ctx, r.targetEndpoint)
```

Comment:
I think it will create a new client on every HTTP request. Is this what we want?

Reply:
I think that's what the readinessProbe does today as well so I would think it should be okay load wise.
https://github.com/openshift/cluster-etcd-operator/blob/master/bindata/etcd/pod.yaml#L207-L211

And makes sense to me that we would want to test out a new connection for the health check but open to hearing why this might be a bad idea.

Reply:
> I think it will create a new client on every HTTP request. Is this what we want?

To start, yes.